torch_np/_ndarray.py

```diff
@@ -433,10 +438,10 @@ def __getitem__(self, index):
     def __setitem__(self, index, value):
         index = _helpers.ndarrays_to_tensors(index)
         index = ndarray._upcast_int_indices(index)
-        value = _helpers.ndarrays_to_tensors(value)
 
-        if isinstance(value, torch.Tensor):
-            value = _util.cast_if_needed(value, self.tensor.dtype)
+        value = normalize_array_like(value)
+        value = _util.cast_if_needed(value, self.tensor.dtype)
```

Comment:
As discussed, let's do this if `type(value) not in {bool,int,float,complex}`, to avoid wrapping scalars into tensors unnecessarily.